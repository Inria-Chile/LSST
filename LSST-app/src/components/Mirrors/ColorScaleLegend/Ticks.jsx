import React, { Component } from 'react'

class ColorScaleLegend extends Component {

    constructor(props) {
        super(props);
        this.state = {
        }
    }

    render() {
        let tickStyle = {
            opacity: 1,
        };
        let textStyle = {
            textAnchor: 'end'
        }
        return (
            <g className="legend axis" transform={"translate("+(this.props.ticksOffset)+", 0)"}>
                <g className="tick" transform={"translate(0,"+(this.props.height-this.props.textOffset)+")"} style={tickStyle}>
                    <line x1={this.props.textOffset-6} y1="0" x2={this.props.textOffset} y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>{this.props.min}</text>
                </g>
                {/* <g className="tick" transform="translate(0,300)" style={tickStyle}>
                    <line x2="6" y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>-2</text>
                </g>
                <g className="tick" transform="translate(0,240.00000000000003)" style={tickStyle}>
                    <line x2="6" y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>-1</text>
                </g>
                <g className="tick" transform="translate(0,180)" style={tickStyle}>
                    <line x2="6" y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>0</text>
                </g>
                <g className="tick" transform="translate(0,120.00000000000001)" style={tickStyle}>
                    <line x2="6" y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>1</text>
                </g>
                <g className="tick" transform="translate(0,59.999999999999986)" style={tickStyle}>
                    <line x2="6" y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>2</text>
                </g> */}
                <g className="tick" transform="translate(0,0)" style={tickStyle}>
                    <line x1={this.props.textOffset-6} y1="0" x2={this.props.textOffset} y2="0"></line>
                    <text dy=".32em" x="9" y="0" style={textStyle}>{this.props.max}</text>
                </g>
            </g>
        );
    }
}


export default ColorScaleLegend;