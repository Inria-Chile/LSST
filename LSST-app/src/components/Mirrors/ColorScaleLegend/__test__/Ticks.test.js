import React from 'react';
import Ticks from '../Ticks';
import renderer from 'react-test-renderer';

describe('Ticks_Test',function(){

    it('renders correctly',()=> {
        const tree = renderer
        .create(<Ticks max={10} min={0} 
            height={250} 
            ticksOffset={0} textOffset={20} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    })
})