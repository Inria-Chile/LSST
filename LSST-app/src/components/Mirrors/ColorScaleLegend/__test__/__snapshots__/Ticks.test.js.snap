// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticks_Test renders correctly 1`] = `
<g
  className="legend axis"
  transform="translate(0, 0)"
>
  <g
    className="tick"
    style={
      Object {
        "opacity": 1,
      }
    }
    transform="translate(0,230)"
  >
    <line
      x1={14}
      x2={20}
      y1="0"
      y2="0"
    />
    <text
      dy=".32em"
      style={
        Object {
          "textAnchor": "end",
        }
      }
      x="9"
      y="0"
    >
      0
    </text>
  </g>
  <g
    className="tick"
    style={
      Object {
        "opacity": 1,
      }
    }
    transform="translate(0,0)"
  >
    <line
      x1={14}
      x2={20}
      y1="0"
      y2="0"
    />
    <text
      dy=".32em"
      style={
        Object {
          "textAnchor": "end",
        }
      }
      x="9"
      y="0"
    >
      10
    </text>
  </g>
</g>
`;
